trade-finance-manager-api/src/v1/rest-mappings/mappings/gef-facilities/mapGefFacilityDates.test.js

```diff
@@ -0,0 +1,202 @@
+
+    expect(result.bankIssueNoticeReceived).toEqual(mockFacility.facilitySnapshot.submittedAsIssuedDate);
+  });
+
+  it('should return coverStartDate as timestamp', () => {
+    const result = mapGefFacilityDates(mockFacility, mockFacilityTfm, MOCK_GEF_DEAL);
+
+    const expected = convertDateToTimestamp(mockFacility.facilitySnapshot.coverStartDate);
+
+    expect(result.coverStartDate).toEqual(expected);
+  });
+
+  it('should return mapped coverEndDate', () => {
+    const result = mapGefFacilityDates(mockFacility, mockFacilityTfm, MOCK_GEF_DEAL);
+
+    const mockCoverEndDate = new Date(mockFacility.facilitySnapshot.coverEndDate);
+
+    const expected = mapCoverEndDate(format(mockCoverEndDate, 'dd'), format(mockCoverEndDate, 'MM'), format(mockCoverEndDate, 'yyyy'), mockFacility);
+
+    expect(result.coverEndDate).toEqual(expected);
+  });
+
+  describe('facility end date feature flag not enabled', () => {
+    beforeAll(() => {
+      jest.mocked(isTfmFacilityEndDateFeatureFlagEnabled).mockReturnValue(false);
+    });
+
+    afterAll(() => {
+      jest.clearAllMocks();
+    });
+
+    it('should not return facility end date fields', () => {
+      const facilityWithEndDate = {
+        facilitySnapshot: {
+          ...mockFacility.facilitySnapshot,
+          isUsingFacilityEndDate: true,
+          facilityEndDate: '2021-12-08T00:00:00.000Z',
+          bankReviewDate: '2023-12-08T00:00:00.000Z',
+        },
+      };
+
+      const result = mapGefFacilityDates(facilityWithEndDate, mockFacilityTfm, MOCK_GEF_DEAL);
+
+      expect(result.isUsingFacilityEndDate).toBeUndefined();
+      expect(result.facilityEndDate).toBeUndefined();
+      expect(result.bankReviewDate).toBeUndefined();
+    });
+  });
+
+  describe('facility end date feature flag enabled', () => {
+    beforeAll(() => {
+      jest.mocked(isTfmFacilityEndDateFeatureFlagEnabled).mockReturnValue(true);
+    });
+
+    afterAll(() => {
+      jest.clearAllMocks();
+    });
+
+    it('should return mapped isUsingFacilityEndDate', () => {
+      const facilityWithEndDate = {
+        facilitySnapshot: {
+          ...mockFacility.facilitySnapshot,
+          isUsingFacilityEndDate: true,
+        },
+      };
+
+      const result = mapGefFacilityDates(facilityWithEndDate, mockFacilityTfm, MOCK_GEF_DEAL);
+
+      expect(result.isUsingFacilityEndDate).toEqual(facilityWithEndDate.facilitySnapshot.isUsingFacilityEndDate);
+    });
+
+    it('should return mapped facilityEndDate if isUsingFacilityEndDate is true', () => {
+      const facilityWithEndDate = {
+        facilitySnapshot: {
+          ...mockFacility.facilitySnapshot,
+          isUsingFacilityEndDate: true,
+          facilityEndDate: '2021-12-08T00:00:00.000Z',
+        },
+      };
+
+      const result = mapGefFacilityDates(facilityWithEndDate, mockFacilityTfm, MOCK_GEF_DEAL);
+      const expected = convertDateToTimestamp(facilityWithEndDate.facilitySnapshot.facilityEndDate);
```

Comment:
Think I'd prefer this to not call another function

```suggestion
      const expected = 1638921600000
```

Or, to make it clear where the value comes from, could have `facilityEndDate = new Date(1638921600000)` somewhere & pass in `facilityEndDate: formatISO(facilityEndDate)` using [date-fns](https://date-fns.org/v3.6.0/docs/formatISO)